stglib/iq.py

```diff
@@ -91,142 +96,174 @@ def read_iq(filnam):
 
-    for var in ["FlowData_Vel_Mean", "FlowData_Vel"]:
-        iq[var] = iq[var] / 1000
+    if ds.attrs["InstrumentType"] == "IQ":
+        ds.attrs["AlongChannelBeamsAngle"] = 25
+        ds.attrs["AcrossChannelBeamsAngle"] = 60
+    else:
+        print("Check and update beam angle for Sontek IQ instrument")
 
-    return iq
+    return xr.decode_cf(ds)
 
 
-def make_beamdist(iq):
+def create_iqbindist(ds):
     """
-    Generate physical coordinates to pair with the logical beamdist coordinates
+    Generate bin distances from transducer along vertical profile for the along (beams 0,1)
+    and across (beams 2,3) bins. Cannot make bindist a dim because it changes by sample due
+    to bin size changing based on water depth.
     """
     for bm in range(4):
         if bm < 2:
-            bdname = "beamdist_0_1"
+            bdname = "bin_along"
         else:
-            bdname = "beamdist_2_3"
-
-        r = range(len(iq[bdname]))
+            bdname = "bin_across"
 
-        time = np.tile(iq["time"], (len(iq[bdname]), 1)).transpose()
-
-        cells = np.zeros(np.shape(iq["Profile_" + str(bm) + "_Vel"]))
+        r = range(len(ds[bdname]))
+        cells = np.zeros(np.shape(ds["Profile_" + str(bm) + "_Vel"]))
         fsdbd = "FlowSubData_PrfHeader_" + str(bm) + "_BlankingDistance"
         fsdcs = "FlowSubData_PrfHeader_" + str(bm) + "_CellSize"
-        for n in range(len(iq["time"])):
-            cells[n, :] = iq[fsdbd][n].values + r * iq[fsdcs][n].values
-        iq["cells_" + str(bm)] = xr.DataArray(cells, dims=("time", bdname))
-        iq["time_" + str(bm)] = xr.DataArray(time, dims=("time", bdname))
-        iq = iq.set_coords(["cells_" + str(bm), "time_" + str(bm)])
-
-    return iq
-
-
-def make_iq_plots(iq, directory="", savefig=False):
-    """
-    Make IQ turnaround plots
-    """
-
-    plt.figure(figsize=(11, 8.5))
+        for n in range(len(ds["time"])):
+            # blanking distance + 1*bin_size = center of first bin
+            # blanking distance + N*bin_size = center of N bin
+            # due to 0 index, have to add 1 additional bin size to equation below
+            # blanking distance + 1*binsize + (bin_along(or bin_across)*bin_size) = center of each bin
+            cells[n, :] = (
+                ds[fsdbd][n].values + (r * ds[fsdcs][n].values) + (ds[fsdcs][n].values)
+            )
+        ds["Profile_" + str(bm) + "_bindist"] = xr.DataArray(
+            cells, dims=("time", bdname)
+        )
 
-    for n, var in enumerate(
-        ["FlowData_Depth", "FlowData_Vel_Mean", "FlowData_Flow"], start=1
-    ):
-        plt.subplot(3, 1, n)
-        plt.plot(iq["time"], iq[var])
-        plt.ylabel(var + " [" + iq[var].attrs["units"] + "]")
+        ds["Profile_" + str(bm) + "_bindist"].attrs.update(
+            {
+                "units": "m",
+                "long_name": "bin (center) distance from transducer",
+                "positive": "%s" % ds.attrs["orientation"],
```

Comment:
This isn't a requirement, but this and others below might be better suited as f-strings.

Reply:
Do you have an example of how to do this? I looked online and examples I'm seeing aren't working with attributes.

Reply:
Because I'm using attrs.update(), it will not let me use f-string with a ds attribute as the string to fill in. For it to work, I have to create a variable for that attribute before using attrs.update(). I can do this but it will be adding extra lines. What do you think?

Reply:
So you are saying `ds["Profile_" + str(bm) + "_bindist"].attrs.update("positive": f"{ds.attrs['orientation']}")` does not work? And really, since this is a simple one with no substitutions it could be simplified to `ds["Profile_" + str(bm) + "_bindist"].attrs.update("positive": ds.attrs["orientation"])`

Reply:
Yeah, I would need to create a variable from that attribute and called the variable instead of the attribute in order for it to have worked. I think it had to do with attrs.update(). Which I can do this if that is preferred. 

And yes, you are right I can also just use `ds["Profile_" + str(bm) + "_bindist"].attrs.update({"positive": ds.attrs["orientation"]})`

Reply:
With the help of Steve, I figured out why it was not working for me. It's because I was using the double quotes when calling the attribute (ds.attrs["orientation"]). I thought I tried it that way already but I guess not. Anyways, I was able to change those to f strings.